File: iosApp/iosApp/ui/textStyles.swift
import UIKit
import SwiftUI
import Shared

struct TextStyle {
    let fontSize: CGFloat
    let fontWeight: Font.Weight
    let lineHeight: CGFloat
    let color: Color?

    init(fontSize: CGFloat, fontWeight: Font.Weight, lineHeight: CGFloat, color: Color? = nil) {
        self.fontSize = fontSize
        self.fontWeight = fontWeight
        self.lineHeight = lineHeight
        self.color = color
    }
}


struct TextStyles {
    static let topLabelText = TextStyle(
        fontSize: 14,
        fontWeight: .semibold,
        lineHeight: 16.71
    )
    
    static let hint = TextStyle(
        fontSize: 16,
        fontWeight: .regular,
        lineHeight: 19.09
    )
    
    static let error = TextStyle(
        fontSize: 12,
        fontWeight: .regular,
        lineHeight: 14.31,
        color: .rebel
    )
    static let bigTitle = TextStyle(
       fontSize: 64,
       fontWeight: .bold,
       lineHeight: 60
    )
}

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/components/TextStyles.kt
package com.wisemen.chargehub.ui.components

import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp
import com.wisemen.chargehub.ui.theme.Colors

object TextStyles {

    val topLabelText = TextStyle(
        fontSize = 14.sp,
        fontWeight = FontWeight.W600,
        lineHeight = 16.71.sp
    )

    val hint = TextStyle(
        fontSize = 16.sp,
        fontWeight = FontWeight.W400,
        lineHeight = 19.09.sp,
    )

    val error = TextStyle(
        fontSize = 12.sp,
        fontWeight = FontWeight.W400,
        lineHeight = 14.31.sp,
        color = Colors.rebel
    )

    val nextReservationText = TextStyle(
        fontSize = 17.sp,
        fontWeight = FontWeight.W600,
        lineHeight = 25.sp,
        color = Colors.white
    )

    val nextReservationHeader = TextStyle(
        fontSize = 10.sp,
        fontWeight = FontWeight.W400,
        lineHeight = 20.29.sp,
        color = Color(0XFF999999)
    )

    val nextReservationCardTitle = TextStyle(
        fontSize = 17.sp,
        fontWeight = FontWeight.W600,
        lineHeight = 22.sp,
        color = Colors.acid
    )

    val nextReservationCardSubTitle = TextStyle(
        fontSize = 13.sp,
        fontWeight = FontWeight.W400,
        lineHeight = 18.sp,
        color = Colors.white
    )

    val timeSlotText = TextStyle(
        fontSize = 17.sp,
        fontWeight = FontWeight.W400,
        lineHeight = 11.93.sp,
        color = Colors.blackPearl
    )

    val timeSlotHeader = TextStyle(
        fontSize = 10.sp,
        fontWeight = FontWeight.W400,
        lineHeight = 20.29.sp,
        color = Colors.mist
    )

    val timeSlotCardTitle = TextStyle(
        fontSize = 17.sp,
        fontWeight = FontWeight.W600,
        lineHeight = 20.29.sp
    )

    val timeSlotDateDay = TextStyle(
        fontSize = 21.sp,
        fontWeight = FontWeight.W700,
        lineHeight = 25.sp
    )

    val timeSlotDateMonth = TextStyle(
        fontSize = 14.sp,
        fontWeight = FontWeight.W400,
        lineHeight = 16.71.sp
    )

}